desktop-exporter/app/components/sidebar-view/sidebar.tsx

```diff
@@ -13,36 +26,73 @@ type SidebarProps = {
 
 export function Sidebar(props: SidebarProps) {
   let sidebarColour = useColorModeValue("gray.50", "gray.700");
+  let iconColour = useColorModeValue("white", "pink.900");
+  let { colorMode, toggleColorMode } = useColorMode();
   let { isFullWidth, toggle, traceSummaries } = props;
 
-  let sidebarWidth = "70px";
-  let buttonIcon = <ArrowRightIcon />;
-  let traceList = <></>;
-
   if (isFullWidth) {
-    sidebarWidth = "350px";
-    buttonIcon = <ArrowLeftIcon />;
-    traceList = <TraceList traceSummaries={traceSummaries} />;
+    return (
+      <Flex
+        backgroundColor={sidebarColour}
+        flexShrink="0"
+        direction="column"
+        transition="width 0.2s ease-in-out"
+        width={sidebarFullWidth}
+      >
+        <Flex
+          height="100px"
+          justifyContent="flex-end"
+          alignItems="center"
+        >
+          <IconButton
+            aria-label="Toddle Colour Mode"
+            color={iconColour}
+            colorScheme="pink"
+            icon={colorMode === "light" ? <MoonIcon /> : <SunIcon />}
+            marginEnd="16px"
+            onClick={toggleColorMode}
+          />
+          <IconButton
+            aria-label="Collapse Sidebar"
+            color={iconColour}
+            colorScheme="pink"
+            icon={<ArrowLeftIcon />}
+            marginEnd="16px"
+            onClick={toggle}
+          />
+        </Flex>
+        <TraceList traceSummaries={traceSummaries} />
+      </Flex>
+    );
   }
 
   return (
     <Flex
-      bgColor={sidebarColour}
+      alignItems="center"
+      backgroundColor={sidebarColour}
       flexShrink="0"
       direction="column"
       transition="width 0.2s ease-in-out"
-      width={sidebarWidth}
+      width={sidebarCollapsedWidth}
     >
-      <Flex justifyContent="flex-end">
-        <IconButton
-          aria-label="Expand Sidebar"
-          colorScheme="pink"
-          icon={buttonIcon}
-          margin="15px"
-          onClick={toggle}
-        />
-      </Flex>
-      {traceList}
+      <IconButton
+        aria-label="Expand Sidebar"
+        color={iconColour}
+        colorScheme="pink"
+        icon={<ArrowRightIcon />}
+        marginTop="16px"
+        onClick={toggle}
+      />
+      <IconButton
+        aria-label="Toddle Colour Mode"
+        color={iconColour}
+        colorScheme="pink"
+        height="40px"
+        icon={colorMode === "light" ? <MoonIcon /> : <SunIcon />}
+        marginTop="16px"
+        onClick={toggleColorMode}
+        width="40px"
+      />
```

Comment:
Potentially these could be returned in a fragment as `SidebarButtons`

Reply:
Done!